### PR TITLE
Add use case and update format

### PR DESCRIPTION
**Pull request by KSHan29**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -312,49 +307,69 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-2. NUScheduler adds the event and displays the updated schedule of events
+1. User requests to add an event to the schedule.
+2. NUScheduler adds the event and displays the updated schedule of events.
 
    Use case ends.
 
 **Extensions**
 
-* 1a. The starting date or the ending date is not provided.
+* 1a. The Title, starting date or the ending date is not provided.
```
Comment (lesterong) [inline docs/DeveloperGuide.md:339]: Not sure if we are simplifying this, but if we are not, I think `name` might be a better word to match our User Guide.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -312,49 +307,69 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends.
 
-      Use case resumes at step 2.
+**Extensions**
+
+* 2a. The given index is invalid.
+
+    * 2a1. NUScheduler shows an error message.
+
+      Use case ends.
```
Comment (lesterong) [inline docs/DeveloperGuide.md:326]: Should it be `Use case resumes at step 1` instead?

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -312,49 +307,69 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* 1b. The given time period is invalid.
+      Use case ends.
+    
+* 2a. The schedule is empty.
 
-    * 1b1. NUScheduler shows an error message.
+  Use case ends.
 
-      Use case ends.
+* 2b. There are no events in the next specified days.
```
Comment (lesterong) [inline docs/DeveloperGuide.md:370]: Is similar to `2a`?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (ichigh0st) [reply docs/DeveloperGuide.md:326]: wouldn't resuming at step 1 be the same as the use case ending?

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (KSHan29) [reply docs/DeveloperGuide.md:370]: Yea I think I agree its quite similar but I interpreted 2a as meaning the entire schedule being empty so I added another extension where nothing is shown. Maybe I could combine both into `There are no events in the next specified days` which covers both situation?

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (KSHan29) [reply docs/DeveloperGuide.md:326]: I think I made a mistake for this edit. The original was to resume from step 2. I think this was meant to signify that user can try to request deleting profile again but does not have to go through step 1 (List all profiles to check for index). I will revert this section?

Comment (lesterong) [reply docs/DeveloperGuide.md:326]: I feel like there should be some way to show that the user can just try to request deleting again and again until it's successful, that's why I think `Use case ends.` may not be very suitable. But I do agree that saying resuming at step 1 sounds a little odd.

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (lesterong) [reply docs/DeveloperGuide.md:370]: I think that's a good suggestion, and after your comment, I think the current more verbose version is fine as well. What do the rest think?

**Review by lesterong — APPROVED** [review]
Looks good! Thanks 👍🏼

**Pull request merged**